core/template.js

```diff
@@ -1028,40 +1028,43 @@ var Template = Montage.specialize( /** @lends Template# */ {
                     }   
                 }
             }
-            result.elementIds = argumentsElementIds;
-            result.elementIdsCollisions = argumentElementsCollisionTable;
-
-            // Expand objects.
-            argumentsSerialization = templateArgumentProvider
-                .getTemplateArgumentSerialization(argumentsElementIds);
-
-            argumentsSerialization.renameElementReferences(
-                argumentElementsCollisionTable);
-
-            // When merging the serializations we need to resolve any template
-            // property alias that comes from the arguments, for instance, the
-            // argument could be referring to @table:cell in its scope when in
-            // this scope (the serialization1) it is aliased to
-            // @repetition:iteration. To do this we ask the argument provider
-            // to resolve the template property for us.
-            // This approach works because the arguments serialization is
-            // created assuming that template properties are just like any other
-            // label and are considered external objects.
-            willMergeObjectWithLabel = function (label) {
-                if (label.indexOf(":") > 0) {
-                    return templateArgumentProvider
-                        .resolveTemplateArgumentTemplateProperty(label);
-                }
-            };
 
-            objectsCollisionTable = serialization.mergeSerialization(
-                argumentsSerialization, {
-                    willMergeObjectWithLabel: willMergeObjectWithLabel
-                });
-            this.objectsString = serialization.getSerializationString();
+            if (templateArgumentProvider._ownerDocumentPart) {
+                result.elementIds = argumentsElementIds;
+                result.elementIdsCollisions = argumentElementsCollisionTable;
+
+                // Expand objects.
+                argumentsSerialization = templateArgumentProvider
+                    .getTemplateArgumentSerialization(argumentsElementIds);
+
+                argumentsSerialization.renameElementReferences(
+                    argumentElementsCollisionTable);
+
+                // When merging the serializations we need to resolve any template
+                // property alias that comes from the arguments, for instance, the
+                // argument could be referring to @table:cell in its scope when in
+                // this scope (the serialization1) it is aliased to
+                // @repetition:iteration. To do this we ask the argument provider
+                // to resolve the template property for us.
+                // This approach works because the arguments serialization is
+                // created assuming that template properties are just like any other
+                // label and are considered external objects.
+                willMergeObjectWithLabel = function (label) {
```

Comment:
Probably, but that is not the purpose of this Pr. It can seem like a huge change but actually the only change is here: https://github.com/montagejs/montage/pull/1931/files/bae328877718cf96393c0e0dc7eb58f83a025fd0#diff-64c6b030384ce8b35a80e02402a5b839R1032
A condition was missing and it was impossible to inject a template within a slot component.

Reply:
I can create a ticket for this matter.